pkg/reconciler/common/prune_test.go

```diff
@@ -324,3 +324,87 @@ func TestConfigChange(t *testing.T) {
 		assert.Error(t, err, "a new cronjob expected")
 	}
 }
+
+// test the cronjob creation when NodeSelector and Toleration changes
+func TestNodeSelectorChange(t *testing.T) {
+	keep := uint(3)
+
+	defaultPrune := &v1alpha1.Prune{
+		Resources: []string{"pipelinerun"},
+		Keep:      &keep,
+		Schedule:  scheduleCommon,
+	}
+	nodeSelectorConfig := v1alpha1.Config{
+		NodeSelector: map[string]string{
+			"foo": "bar",
+		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:      "foo",
+				Operator: "equals",
+				Value:    "bar",
+				Effect:   "noSchedule",
+			},
+		},
+	}
+	nodeSelectorEmptyConfig := v1alpha1.Config{}
+
+	config1 := &v1alpha1.TektonConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "config",
+		},
+		Spec: v1alpha1.TektonConfigSpec{
+			Profile:    "all",
+			Pruner:     *defaultPrune,
+			CommonSpec: v1alpha1.CommonSpec{TargetNamespace: "openshift-pipelines"},
+			Config:     nodeSelectorConfig,
+		},
+	}
+
+	config2 := &v1alpha1.TektonConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "config",
+		},
+		Spec: v1alpha1.TektonConfigSpec{
+			Profile:    "all",
+			Pruner:     *defaultPrune,
+			CommonSpec: v1alpha1.CommonSpec{TargetNamespace: "openshift-pipelines"},
+			Config:     nodeSelectorEmptyConfig,
+		},
+	}
+	client := fake.NewSimpleClientset(
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "openshift-pipelines"}},
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "openshift-api"}},
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "openshift-api-url"}},
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "kube-system"}},
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "kube-api"}},
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "ns-one"}},
+	)
+	os.Setenv(JobsTKNImageName, "some")
```

Comment:
usually you would want to do defer function to unset the env (with https://pkg.go.dev/gotest.tools/v3/env for example)